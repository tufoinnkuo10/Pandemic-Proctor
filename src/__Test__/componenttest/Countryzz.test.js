import React from 'react';
import renderer from 'react-test-renderer';
import Countries from '../../Components/Countryzz';

const mockedData = {
  Cameroon: {
    new_cases: 604,
    death: 21,
    today_confirmed: 106794,
  },
  Benin: {
    new_cases: 4,
    death: 0,
    today_confirmed: 24850,
  },
};

const mockedTotalCases = {
  today_confirmed: 33868,
  today_new_confirmed: 6753375,
  today_deaths: 575238,
};

describe('Countries', () => {
  it('renders ! ', () => {
    const tree = renderer
      .create(
        <Countries
          countries={mockedData}
          total={mockedTotalCases}
          loading={false}
          handleClick={() => 7 + 5}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
