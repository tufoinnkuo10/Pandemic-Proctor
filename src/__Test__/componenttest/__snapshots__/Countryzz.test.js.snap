// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries renders !  1`] = `
<div>
  <div
    className="header-wrapper"
  >
    <div
      className="world-header"
    >
      <i
        alt=""
        className="world-map"
        src="worldmapss.png"
      />
    </div>
    <div
      className="world-info"
    >
      <span
        className="world-cases"
      >
        Roughly 
      </span>
      <span
        className="world-cases-value"
      >
         
        33.868
      </span>
      <span
        className="world-cases"
      >
        Covid world Cases 
      </span>
      <svg
        className="icon-virus"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M483.55,227.55H462c-50.68,0-76.07-61.27-40.23-97.11L437,115.19A28.44,28.44,0,0,0,396.8,75L381.56,90.22c-35.84,35.83-97.11,10.45-97.11-40.23V28.44a28.45,28.45,0,0,0-56.9,0V50c0,50.68-61.27,76.06-97.11,40.23L115.2,75A28.44,28.44,0,0,0,75,115.19l15.25,15.25c35.84,35.84,10.45,97.11-40.23,97.11H28.45a28.45,28.45,0,1,0,0,56.89H50c50.68,0,76.07,61.28,40.23,97.12L75,396.8A28.45,28.45,0,0,0,115.2,437l15.24-15.25c35.84-35.84,97.11-10.45,97.11,40.23v21.54a28.45,28.45,0,0,0,56.9,0V462c0-50.68,61.27-76.07,97.11-40.23L396.8,437A28.45,28.45,0,0,0,437,396.8l-15.25-15.24c-35.84-35.84-10.45-97.12,40.23-97.12h21.54a28.45,28.45,0,1,0,0-56.89ZM224,272a48,48,0,1,1,48-48A48,48,0,0,1,224,272Zm80,56a24,24,0,1,1,24-24A24,24,0,0,1,304,328Z"
        />
      </svg>
    </div>
  </div>
  <form
    className="form"
  >
    <input
      className="input-search"
      onChange={[Function]}
      placeholder="SEARCH: Ctry name"
      type="text"
    />
  </form>
  <ul
    className="countries-list"
  >
    <li
      className="countries"
    >
      <button
        className="country"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="info-icon"
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.293 7.707 13.586 12l-4.293 4.293 1.414 1.414L16.414 12l-5.707-5.707z"
          />
        </svg>
        <img
          alt="map"
          className="country-map"
          height="100"
          src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/cm/vector.svg"
          width="100"
        />
        <div
          className="country-data"
        >
          <h1
            className="country-name"
          >
            CAMEROON
          </h1>
          <span
            className="total-confirmed"
          >
            106.794
          </span>
        </div>
      </button>
    </li>
    <li
      className="countries"
    >
      <button
        className="country"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="info-icon"
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.293 7.707 13.586 12l-4.293 4.293 1.414 1.414L16.414 12l-5.707-5.707z"
          />
        </svg>
        <img
          alt="map"
          className="country-map"
          height="100"
          src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/be/vector.svg"
          width="100"
        />
        <div
          className="country-data"
        >
          <h1
            className="country-name"
          >
            BENIN
          </h1>
          <span
            className="total-confirmed"
          >
            24.850
          </span>
        </div>
      </button>
    </li>
  </ul>
</div>
`;
